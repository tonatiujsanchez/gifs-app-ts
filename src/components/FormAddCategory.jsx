import { useState } from 'react';



export const FormAddCategory= ({ onAddCategory }) => {


    const [ inputValue, setInputValue ] = useState('Valorant')

    
    const onInputChange = ({ target }) => {
        setInputValue( target.value )
    }


    const onSubmit = ( e) => {
        e.preventDefault()

        if(inputValue.trim() === ''){ return }

        onAddCategory(inputValue.trim())
        setInputValue('')
        
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ onInputChange }
                placeholder="Buscar Gif"
            />
        </form>
    )
}
