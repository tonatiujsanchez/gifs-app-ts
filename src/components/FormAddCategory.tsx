import { ChangeEvent, FC, FormEvent, useState } from 'react';


interface Props {
    onAddCategory: ( value:string )=> void
}

export const FormAddCategory: FC<Props> = ({ onAddCategory }) => {


    const [ inputValue, setInputValue ] = useState<string>('Valorant')

    
    const onInputChange = ({ target }:ChangeEvent<HTMLInputElement>) => {
        setInputValue( target.value )
    }


    const onSubmit = ( e:FormEvent<HTMLFormElement>) => {
        e.preventDefault()

        if(inputValue.trim() === ''){ return }

        onAddCategory(inputValue.trim())
        setInputValue('')
        
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ onInputChange }
                placeholder="Buscar Gif"
            />
        </form>
    )
}
