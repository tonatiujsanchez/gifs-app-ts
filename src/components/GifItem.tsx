import { FC } from "react"


interface Props {
    title: string
    url: string
}

export const GifItem:FC<Props> = ({ title, url }) => {
    
    
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{ title }</p>
        </div>
    )
}
