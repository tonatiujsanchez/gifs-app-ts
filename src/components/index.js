export * from './FormAddCategory'
export * from './GifGrid'
export * from './GifItem'