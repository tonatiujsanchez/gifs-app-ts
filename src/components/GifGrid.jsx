import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks'

import { GifItem } from './GifItem'




export const GifGrid= ({ category }) => {

    const { images, isLoading } = useFetchGifs( category )
    

    return (
        <>
        <h3>{category}</h3>
        {
            isLoading 
                ?(
                    <div>Cargando...</div>
                ):(
                    <div className='card-grid'>
                        {
                            images.map( ( image  )=> (
                                <GifItem key={image.id} { ...image } />
                            ))
                        }
                    </div>
                )
        }
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
