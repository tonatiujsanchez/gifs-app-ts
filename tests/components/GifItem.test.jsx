const { render, screen } = require("@testing-library/react")
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en el componente GifItem', () => {

    const title="Hola, soy Goku"
    const url="https://giphy.com/gifs/adultswim-adult-swim-toonami-l3vRlt8kty8KKeHni"
 

    test('Debe de hacer match con el snapshot', () => { 
        
        const { container } = render( 
            <GifItem 
                title={ title }
                url={url}
            /> )

        expect( container ).toMatchSnapshot()
    })


    test('Debe de mostrar la imageb con el URL y el ALT inidcado', () => {

        render(  <GifItem title={ title } url={url} /> )

        // screen.debug()

        // expect( screen.getByRole('img').src ).toBe( url )
        // expect( screen.getByRole('img').alt ).toBe( title )
                
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )
    })


    test('Debe de mostrar el titulo en el componente', ()=> {
        render(  <GifItem title={ title } url={url} /> )
        expect( screen.getByText(title) ).toBeTruthy()
    })

})