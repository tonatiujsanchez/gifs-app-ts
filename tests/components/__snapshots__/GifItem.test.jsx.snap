// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola, soy Goku"
      src="https://giphy.com/gifs/adultswim-adult-swim-toonami-l3vRlt8kty8KKeHni"
    />
    <p>
      Hola, soy Goku
    </p>
  </div>
</div>
`;
